MvvmCross/Droid/Droid/Views/MvxAndroidViewPresenter.cs

```diff
@@ -236,39 +240,42 @@ protected Type GetCurrentActivityViewModelType()
             if (attribute.ActivityHostViewModelType == null)
                 attribute.ActivityHostViewModelType = GetCurrentActivityViewModelType();
 
-            Task.Run(async () => {
-                await ShowHostActivity(attribute);
-            }).ContinueWith((result) => {
-                Mvx.Resolve<IMvxMainThreadDispatcher>().RequestMainThreadAction(() => {
-                    if (CurrentActivity.FindViewById(attribute.FragmentContentId) == null)
-                        throw new NullReferenceException("FrameLayout to show Fragment not found");
+            var currentHostViewModelType = GetCurrentActivityViewModelType();
+            if (attribute.ActivityHostViewModelType != currentHostViewModelType)
+            {
+                _pendingRequest = request;
```

Comment:
The problem with that is that we need to implement this in the Activity, while this solution keeps everything in the presenter. Also we wouldn't be able to send a ViewModel instance in the Intent.